packages/fe-fpm-writer/test/unit/view.test.ts

```diff
@@ -0,0 +1,79 @@
+import { create, Editor } from 'mem-fs-editor';
+import { create as createStorage } from 'mem-fs';
+import { join } from 'path';
+import { generateCustomView } from '../../src';
+import { CustomView } from '../../src/view/types';
+import * as manifest from './sample/column/webapp/manifest.json';
+
+const testDir = join(__dirname, 'sample/view');
+
+describe('CustomView', () => {
+    let fs: Editor;
+    const customView: CustomView = {
+        target: 'sample',
+        key: 'viewKey',
+        label: 'viewLabel',
+        name: 'NewCustomView',
+        folder: 'extensions/custom'
+    };
+    const expectedFragmentPath = join(testDir, 'webapp', customView.folder!, `${customView.name}.fragment.xml`);
+    beforeEach(() => {
+        fs = create(createStorage());
+        fs.delete(testDir);
+        fs.write(join(testDir, 'webapp/manifest.json'), JSON.stringify(manifest));
+    });
+
+    test('only mandatory properties', () => {
+        //sut
+        generateCustomView(testDir, customView, fs);
+        const updatedManifest: any = fs.readJSON(join(testDir, 'webapp/manifest.json'));
+        const extension = updatedManifest['sap.ui5']['extends'];
+        const views = updatedManifest['sap.ui5']['routing']['targets']['sample']['options']['settings']['views'];
+        expect(extension).not.toBeDefined();
+        expect(views).toMatchSnapshot();
+        expect(fs.read(expectedFragmentPath)).toMatchSnapshot();
+    });
+
+    test('with new handler', () => {
+        //sut
+        generateCustomView(testDir, { ...customView, key: 'customViewKey', eventHandler: true }, fs);
+        const updatedManifest: any = fs.readJSON(join(testDir, 'webapp/manifest.json'));
+        const extension = updatedManifest['sap.ui5']['extends']['extensions'];
+        const views = updatedManifest['sap.ui5']['routing']['targets']['sample']['options']['settings']['views'];
+        expect(extension).toMatchSnapshot();
+        expect(views).toMatchSnapshot();
```

Comment:
That's the problem of snapshot testing, that you never look at the content...

Reply:
Still snapshots, but sympotom should be addressed now. If approach is used and preferred, I could add direct tests.

Reply:
What is the problem of snapshot testing? I don't understand the problem here.

Reply:
@tobiasqueck Not especially here, but it is my general concern - but we hade this discussion already some times back.